doc/functions.tsx

```diff
@@ -22,27 +22,34 @@ import { type Child, isDeprecated, take } from "./utils.ts";
 
 export function DocFunctionSummary({
   children,
+  typeParams,
```

Comment:
you could do `...props` here and then do `{...props}` down below.  With one property being passed it seemed silly, but with multiple now, spreading makes it cleaner I think.

Reply:
hm, we used to do that for markdownContext, however i reworked that to not do that because spread is apparently not the fastest of things, especially if its happening _a lot_.
Another idea i had was make typeParams part of MarkdownContext, but that didnt really feel right

Reply:
I would be interested to hear about any performance issues with spread.  When using "traditional" source, there isn't actually any spreading going on at runtime [see swc playground](https://play.swc.rs/?version=1.3.4&code=H4sIAAAAAAAAAz2OwQ6CMBBE7036D3NDiYE7NI3iZxgPiEVqpCXLkpA0%2FXerBuf0dia7O2We4%2FicVwzISynsOHlihDRG9ORHZBOZtuOslkIKs37jfnEdW%2B9w2oVusK87GXdAURRkZo5VWt%2FcCucPqZnJuoeu0Vb4cY3bhpdr3CNIgSQyvJCDahC2e%2Fr%2FI6qy0alIfAO7YB7wtgAAAA%3D%3D&config=H4sIAAAAAAAAA0WOSwrDMAxE76J1FsWLLnyHHsK4SnDxD0mBGuO7VzYp2Q0zo6fp8GEPtkN1xEhTccvivmBBWkX2FKrABsJqCZ04VDs6ULSBbB7GaBpLYQS7u8i4QQo57G2yfEmVkPmOXD7ivzmUlcr7nEZf7xbzCeNmXHeBX1dxbfgBaADhoLcAAAA%3D) it is only when you are using the newer jsx-runtime.  But having said that, V8 is [pretty darn optimised](https://v8.dev/blog/spread-elements) when it comes to spreading.

All that being said, `MarkdownContext` is actually a misnomer here, because it is more a doc context we are passing around, as some things other than the markdown stuff use it.  Maybe just `Context` then, with an optional `typeParams`?

Reply:
> All that being said, `MarkdownContext` is actually a misnomer here, because it is more a doc context we are passing around

yes, i thought so too, but then i checked and noticed that the `Markdown` component actually uses all its properties, so its not much of a misnomer in the end. 
anyhow, yea it might make more sense to just make it more generic and throw typeparams in there

Reply:
> I would be interested to hear about any performance issues with spread. When using "traditional" source, there isn't actually any spreading going on at runtime [see swc playground](https://play.swc.rs/?version=1.3.4&code=H4sIAAAAAAAAAz2OwQ6CMBBE7036D3NDiYE7NI3iZxgPiEVqpCXLkpA0%2FXerBuf0dia7O2We4%2FicVwzISynsOHlihDRG9ORHZBOZtuOslkIKs37jfnEdW%2B9w2oVusK87GXdAURRkZo5VWt%2FcCucPqZnJuoeu0Vb4cY3bhpdr3CNIgSQyvJCDahC2e%2Fr%2FI6qy0alIfAO7YB7wtgAAAA%3D%3D&config=H4sIAAAAAAAAA0WOSwrDMAxE76J1FsWLLnyHHsK4SnDxD0mBGuO7VzYp2Q0zo6fp8GEPtkN1xEhTccvivmBBWkX2FKrABsJqCZ04VDs6ULSBbB7GaBpLYQS7u8i4QQo57G2yfEmVkPmOXD7ivzmUlcr7nEZf7xbzCeNmXHeBX1dxbfgBaADhoLcAAAA%3D) it is only when you are using the newer jsx-runtime. But having said that, V8 is [pretty darn optimised](https://v8.dev/blog/spread-elements) when it comes to spreading.

well, i brought it up to divy when there was talk about spread being a perf problem in cli, so asked him about it in for the markdowncontext case and said it would be better to not constantly spread & unspread. though regardless if it did do anything or not, i would have changed it to not constantly spread & unspread because it was just bothering me 😅

Reply:
> yes, i thought so too, but then i checked and noticed that the `Markdown` component actually uses all its properties, so its not much of a misnomer in the end.
> anyhow, yea it might make more sense to just make it more generic and throw typeparams in there

Just because `Markdown` uses all the context, doesn't mean it is `MarkdownContext`.  😄 yes, that is where it originated, but now it has become bigger than its name.  I do think we should just embiggen its scope and rename it.

Reply:
sure